integration/integration.go

```diff
@@ -351,3 +359,62 @@ func writeToDisk(
 
 	return nil
 }
+
+func newShardsRange(from, to uint32) []uint32 {
+	var ids []uint32
+	for i := from; i <= to; i++ {
+		ids = append(ids, i)
+	}
+	return ids
+}
+
+func newClusterShardsRange(from, to uint32) shard.Shards {
+	return shard.NewShardsWithIDs(newShardsRange(from, to))
+}
+
+func newClusterEmptyShardsRange() shard.Shards {
+	return shard.NewShardsWithIDs(nil)
+}
+
+func waitUntilHasBootstrappedShardsExactly(
+	db storage.Database,
+	shards []uint32,
+) {
+	for {
+		if hasBootstrappedShardsExactly(db, shards) {
+			return
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func hasBootstrappedShardsExactly(
+	db storage.Database,
+	shards []uint32,
+) bool {
+	for _, namespace := range db.Namespaces() {
+		expect := make(map[uint32]struct{})
```

Comment:
Doesn't look like `expect` changes, can we initialize it once and reuses it in the loop?